File: TheepotAndroid/app/src/main/kotlin/nl/bsodetheepot/mobile/data/models/Maaltijdlijst.kt
package nl.bsodetheepot.mobile.data.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.DayOfWeek
import java.time.LocalDate

@Serializable
enum class Dag {
    @SerialName("maandag") MAANDAG,
    @SerialName("dinsdag") DINSDAG,
    @SerialName("woensdag") WOENSDAG,
    @SerialName("donderdag") DONDERDAG,
    @SerialName("vrijdag") VRIJDAG;

    val label: String get() = name.lowercase().replaceFirstChar { it.uppercase() }

    companion object {
        /** Mapt een datum naar de bijbehorende weekdag, of `null` in het weekend. */
        fun vanWeekdag(datum: LocalDate): Dag? = when (datum.dayOfWeek) {
            DayOfWeek.MONDAY -> MAANDAG
            DayOfWeek.TUESDAY -> DINSDAG
            DayOfWeek.WEDNESDAY -> WOENSDAG
            DayOfWeek.THURSDAY -> DONDERDAG
            DayOfWeek.FRIDAY -> VRIJDAG
            else -> null
        }
    }
}

@Serializable
data class MaaltijdLocatie(val id: String, val naam: String, val actief: Boolean)

/** Tabel `maaltijd_standaard_kinderen`. */
@Serializable
data class MaaltijdStandaardKind(
    val id: String,
    @SerialName("locatie_id") val locatieId: String,
    val naam: String,
    val bijzonderheden: String? = null,
    val dag: Dag,
    val volgorde: Int,
)

/** Tabel `maaltijd_weken`. */
@Serializable
data class MaaltijdWeek(
    val id: String,
    @SerialName("locatie_id") val locatieId: String,
    val maand: String,
    @SerialName("week_start") val weekStart: String,
)

/** Tabel `maaltijd_registraties`. */
@Serializable
data class MaaltijdRegistratie(
    val id: String,
    @SerialName("week_id") val weekId: String,
    val dag: Dag,
    val naam: String,
    val bijzonderheden: String? = null,
    @SerialName("wat_gegeten") val watGegeten: String? = null,
    val aanwezig: Boolean = true,
    @SerialName("is_extra") val isExtra: Boolean = false,
    val volgorde: Int = 0,
)

File: TheepotMobile/TheepotMobile/Models/Maaltijdlijst.swift
import Foundation

enum Dag: String, Codable, CaseIterable {
    case maandag, dinsdag, woensdag, donderdag, vrijdag

    var label: String { rawValue.capitalized }
}

/// Tabel `maaltijd_standaard_kinderen`.
struct StandaardKind: Codable, Identifiable {
    let id: String
    let locatieId: String
    let naam: String
    let bijzonderheden: String?
    let dag: Dag
    let volgorde: Int

    enum CodingKeys: String, CodingKey {
        case id
        case locatieId = "locatie_id"
        case naam, bijzonderheden, dag, volgorde
    }
}

/// Tabel `maaltijd_weken`.
struct MaaltijdWeek: Codable, Identifiable {
    let id: String
    let locatieId: String
    var maand: String
    let weekStart: String

    enum CodingKeys: String, CodingKey {
        case id
        case locatieId = "locatie_id"
        case maand
        case weekStart = "week_start"
    }
}

/// Tabel `maaltijd_registraties` — hier wordt "meegegeten" (aanwezig) getoggeld.
struct MaaltijdRegistratie: Codable, Identifiable {
    let id: String
    let weekId: String
    let dag: Dag
    var naam: String
    var bijzonderheden: String?
    var watGegeten: String?
    var aanwezig: Bool
    var isExtra: Bool
    var volgorde: Int

    enum CodingKeys: String, CodingKey {
        case id
        case weekId = "week_id"
        case dag, naam, bijzonderheden
        case watGegeten = "wat_gegeten"
        case aanwezig
        case isExtra = "is_extra"
        case volgorde
    }
}

/// Tabel `maaltijd_locaties`.
struct MaaltijdLocatie: Codable, Identifiable, Hashable {
    let id: String
    let naam: String
    let actief: Bool
}
